presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSplitSource.java

```diff
@@ -83,6 +84,24 @@ public IcebergSplitSource(
                         .iterator());
     }
 
+    public IcebergSplitSource(
+            ConnectorSession session,
+            IncrementalAppendScan scan,
+            TupleDomain<IcebergColumnHandle> metadataColumnConstraints)
```

Comment:
```suggestion
    public <T> IcebergSplitSource(
            ConnectorSession session,
            Scan<T, FileScanTask, CombinedScanTask> tableScan,
            TupleDomain<IcebergColumnHandle> metadataColumnConstraints)
```

I wonder if we can use a unified constructor like this for both `TableScan` and `IncrementalAppendScan`?